src/pages/Registration.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+import "./Registration.css";
+
+interface FormData {
+    username: string;
+    email: string;
+    password: string;
+    repeat_password: string;
+    avatar?: File;
+}
+
+export const FormRegistration = () => {
+    const schema = yup.object().shape({
+        username: yup.string().min(4).required("username is Required!"),
+        email: yup.string().email().required("email is Required!"),
+        password: yup.string().min(4).max(20).required(),
+        repeat_password: yup.string().oneOf([yup.ref("password"), undefined], "passwords don't match!").required(),
+    });
+
```

Comment:
please put schema out of component, if component big(more than 100 stings) better move schema in separate file

Reply:
corrected